cluster-autoscaler/processors/provreq/injector.go

```diff
@@ -29,26 +29,33 @@ import (
 	provreqconditions "k8s.io/autoscaler/cluster-autoscaler/provisioningrequest/conditions"
 	provreqpods "k8s.io/autoscaler/cluster-autoscaler/provisioningrequest/pods"
 	"k8s.io/autoscaler/cluster-autoscaler/provisioningrequest/provreqclient"
+	"k8s.io/autoscaler/cluster-autoscaler/provisioningrequest/provreqwrapper"
 	"k8s.io/client-go/rest"
 	"k8s.io/klog/v2"
 	"k8s.io/utils/clock"
 )
 
 const (
-	defaultRetryTime = 10 * time.Minute
+	defaultRetryTime = 1 * time.Minute
+	maxBackoffTime   = 10 * time.Minute
+	maxCacheSize     = 1000
```

Comment:
I'm not 100% sure if that'll be sufficient long-term. It doesn't sound like it'll scale very well, since it effectively disables any backoff when there are more than 1k failing ProvReqs - i.e., precisely when a backoff would be most useful for preventing starvation of other requests. 

Can you add a todo to clean up ProvReqs when they're resolved (succeed, time out, or are deleted)?

Reply:
To be clear, I'm OK with merging it as is, but I think we should leave a note that it's a stopgap solution.

Reply:
Done, also removed elements that are provisioned or failed